service/src/main/java/com/kodemon/service/facade/FightFacadeImpl.java

```diff
@@ -79,28 +79,33 @@ public void fightForBadge(UserDTO user, GymDTO gym) {
         fight.setFightTime(fightTime);
         LOG.debug("Storing the result of the fight between User {} and Gym {}. Fight time: {}.", user.getId(), gym.getId(), fightTime);
         trainerFightService.save(beanMappingService.mapTo(fight, TrainerFight.class));
+        return wasChallengerSuccessful;
     }
 
     @Override
-    public void fightWildPokemon(UserDTO user, WildPokemonFightMode mode) {
-        Trainer trainer = beanMappingService.mapTo(user, Trainer.class);
-        Pokemon wildPokemon = pokemonService.generateWildPokemon(null);
+    public boolean fightWildPokemon(UserDTO user, PokemonDTO pokemon, WildPokemonFightMode mode) {
+        Collection<Trainer> trainers = trainerService.findByUserName(user.getUserName());
+        if (trainers.isEmpty())
+            return false;
+        Trainer trainer = trainers.iterator().next();
+        Pokemon wildPokemon = beanMappingService.mapTo(pokemon, Pokemon.class);
         LOG.debug("User {} fighting wild Pokemon in mode {}.", user.getId(), mode);
         Pokemon trainersPokemon = pokemonService.findByTrainer(trainer).get(0);
-        Random rand = new Random();
-        wildPokemon.setLevel(Math.max(1, trainersPokemon.getLevel() - 5 + rand.nextInt(10)));
 
         Pair<Double, Double> fightScore = pokemonFightService.getScorePair(trainersPokemon, wildPokemon);
 
         if (fightScore.getX() > fightScore.getY()) {
             if (mode == WildPokemonFightMode.TRAIN) {
                 pokemonService.levelPokemonUp(trainersPokemon);
+                return true;
             } else if (mode == WildPokemonFightMode.CATCH) {
-                pokemonService.save(wildPokemon);
+                pokemonService.assignTrainerToPokemon(trainer, wildPokemon);
```

Comment:
Oh, ok, I got it, but are you removing the Pokemon afterwards? After unsuccessful fight?